### ISSUE TITLE
Problem with exceptions while deleting site data on exit

### ISSUE THREAD

**Issue by DRYN07** [brave/brave-browser#37810]
I would like to clear every site data on browser exit except the sites I allow to store data but I just can't get this work.

Is there any description about this? In Edge, it is very simple.

![image](https://github.com/brave/brave-browser/assets/29770761/fcbc9869-aa40-4735-871e-a6dbea23f423)

But in Brave it is chaotic because there is a lots of settings for this but I found exceptions only to allow third-party cookies.

![image](https://github.com/brave/brave-browser/assets/29770761/cbf760da-1e6f-4c23-8c7d-702ab3e0d6ac)

![image](https://github.com/brave/brave-browser/assets/29770761/e665b5cc-f745-44e3-a7b6-ba8f3d57cc77)

![image](https://github.com/brave/brave-browser/assets/29770761/dd74ce57-ed86-4504-89e6-02b2748425e9)

![image](https://github.com/brave/brave-browser/assets/29770761/9a791ae3-58c7-45b4-aa20-d551ee11f292)

I would really appreciate any help. Again, I want to delete every data on browser exit, EXCEPT for sites I add to the list.

This is a privacy based browser so I would assume that this would be a very basic function. Although I hope it is just me who miss some settings

**Comment by Varauk** [comment]
Greetings, I had a similar problem and did some testing and maybe found a solution. You *can* specify exceptions, and you were actually already on the correct page brave://settings/content/siteData (which exists in chromium too):
![screenshotBraveExceptions](https://github.com/brave/brave-browser/assets/35373497/9d5160f5-de0d-4147-aecf-f74e5f8001d1)
The real problem (or most likely a bug) is that brave added its own version of "clear on exit" stuff (the "On exit" tab in brave://settings/clearBrowserData which you can access by clicking "Clear browsing data" in the privacy/security menu. This one does NOT exist in base chromium), and this *ignores* the settings I showed above. So when you tick the checkbox for clearing cookies and site data in brave://settings/clearBrowserData, brave will do that and ignore all exceptions in brave://settings/content/siteData.
Small sidenote about managing exceptions in brave://settings/content/siteData: when visiting a site you want to an exception for, you can just click on the "View site information" button left of the URL, "Cookies and site data > Manage on device site data" and then select the URLs/domains you want to allow saving persistent cookies.

tl;dr: Disable the checkbox "Cookies and other site data" in brave://settings/clearBrowserData "On exit", and manage on exit clears in brave://settings/content/siteData

**Comment by sydbat** [comment]
> Greetings, I had a similar problem and did some testing and maybe found a solution. You _can_ specify exceptions, and you were actually already on the correct page brave://settings/content/siteData (which exists in chromium too): ![screenshotBraveExceptions](https://github.com/brave/brave-browser/assets/35373497/9d5160f5-de0d-4147-aecf-f74e5f8001d1) The real problem (or most likely a bug) is that brave added its own version of "clear on exit" stuff (the "On exit" tab in brave://settings/clearBrowserData which you can access by clicking "Clear browsing data" in the privacy/security menu. This one does NOT exist in base chromium), and this _ignores_ the settings I showed above. So when you tick the checkbox for clearing cookies and site data in brave://settings/clearBrowserData, brave will do that and ignore all exceptions in brave://settings/content/siteData. Small sidenote about managing exceptions in brave://settings/content/siteData: when visiting a site you want to an exception for, you can just click on the "View site information" button left of the URL, "Cookies and site data > Manage on device site data" and then select the URLs/domains you want to allow saving persistent cookies.
> 
> tl;dr: Disable the checkbox "Cookies and other site data" in brave://settings/clearBrowserData "On exit", and manage on exit clears in brave://settings/content/siteData

Unfortunately, this does not work. At least in Linux. Never had any problem with chromium-based browsers for this issue. Only Brave. If I choose to delete the data when closing in brave://settings/content/siteData, even though I have a list of exceptions, everything gets logged out on browser close.  And in brave://settings/clearBrowserData "On exit" I have everything unchecked. This is not very secure at all. I have tried to post to the Brave forum for help, but my post is either awaiting moderation for some reason, or has just been deleted. Very helpful.

Looking up this issue, it appears that it has been around for several years, and has no solution. Unless this can be sorted, I will have to look for another browser to replace Chrome (because of Google's greed).

**Comment by Kerobyte** [comment]
I've encountered this issue. On-device site data exceptions are cleared after restarting Brave.

<img width="730" height="686" alt="Image" src="https://github.com/user-attachments/assets/53e8141d-9aa5-41f2-8319-aee60026c0ed" />

Linux
[Brave 1.93.136 (Official Build) (x86_64)]
Chromium: 151.0.7922.137
